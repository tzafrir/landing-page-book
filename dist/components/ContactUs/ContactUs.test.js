'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _ContactUs = require('./ContactUs');

var _ContactUs2 = _interopRequireDefault(_ContactUs);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('ContactUs', function () {
  var props = void 0;
  var mounted = void 0;
  var contactUs = function contactUs() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_ContactUs2.default, props));
    }
    return mounted;
  };
  describe('ContactUs', function () {
    beforeEach(function () {
      props = {
        name: 'name',
        mail: 'mail@gmail.com',
        phone: '5476',
        message: 'This is a new message'
      };
      mounted = undefined;
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_ContactUs2.default, props))).toMatchSnapshot();
    });

    it('always renders a div', function () {
      var divs = contactUs().find('div');

      expect(divs.length).toBeGreaterThan(0);
    });

    it('on change message', function () {
      var $event = { value: 'abc' };
      contactUs().instance().onChangeMessage($event);

      expect(contactUs().state('message')).toEqual('abc');
    });

    it('on change phone', function () {
      var $event = { value: '5476' };
      contactUs().instance().onChangePhone($event);

      expect(contactUs().state('phone')).toEqual('5476');
    });

    it('on change email', function () {
      var $event = { value: 'abc@gmail.com' };
      contactUs().instance().onChangeMail($event);

      expect(contactUs().state('mail')).toEqual('abc@gmail.com');
    });

    it('on change message', function () {
      var $event = { value: 'my name' };
      contactUs().instance().onChangeName($event);

      expect(contactUs().state('name')).toEqual('my name');
    });
  });
});