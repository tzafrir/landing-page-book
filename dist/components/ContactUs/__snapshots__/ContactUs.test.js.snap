// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs ContactUs should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactUs
    apiUrl=""
    loading={false}
    mail="mail@gmail.com"
    mailText="Your Email"
    message="This is a new message"
    messageText="Your Message"
    name="name"
    nameText="Your Name"
    onApiFail={[Function]}
    onApiSuccess={[Function]}
    onSubmit={[Function]}
    phone="5476"
    phoneText="Your Phone"
    showPlaceholder={true}
    showText={false}
    submitButtonAlign="center"
    submitButtonColor="default"
    submitButtonText="Submit"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FormGroup
          className="form-group"
        >
          <Input
            className=""
            icon=""
            iconAlign="left"
            id="contactNameTextLabel"
            label=""
            labelColon={false}
            name=""
            onChange={[Function]}
            placeholder="Your Name"
            size=""
            type="text"
            value="name"
          />
        </FormGroup>,
        <FormGroup
          className="form-group"
        >
          <Input
            className=""
            icon=""
            iconAlign="left"
            id="contactEmailTextLabel"
            label=""
            labelColon={false}
            name=""
            onChange={[Function]}
            placeholder="Your Email"
            size=""
            type="email"
            value="mail@gmail.com"
          />
        </FormGroup>,
        <FormGroup
          className="form-group"
        >
          <Input
            className=""
            icon=""
            iconAlign="left"
            id="contactPhoneTextLabel"
            label=""
            labelColon={false}
            name=""
            onChange={[Function]}
            placeholder="Your Phone"
            size=""
            type="text"
            value="5476"
          />
        </FormGroup>,
        <FormGroup
          className="form-group"
        >
          <Input
            className=""
            icon=""
            iconAlign="left"
            id="contactMessageTextLabel"
            label=""
            labelColon={false}
            name=""
            onChange={[Function]}
            placeholder="Your Message"
            size=""
            type="textarea"
            value="This is a new message"
          />
        </FormGroup>,
        <div
          className="text-center"
        >
          <Button
            active={false}
            ariaLabel=""
            circle={false}
            className=""
            color="default"
            disabled={false}
            loading={false}
            loadingClass="fa fa-circle-o-notch fa-spin"
            onClick={[Function]}
            outline={false}
            rounded={true}
            size=""
            style={Object {}}
            tooltip=""
            type="button"
          >
            Submit
          </Button>
        </div>,
      ],
      "className": "ContactUs",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Input
            className=""
            icon=""
            iconAlign="left"
            id="contactNameTextLabel"
            label=""
            labelColon={false}
            name=""
            onChange={[Function]}
            placeholder="Your Name"
            size=""
            type="text"
            value="name"
          />,
          "className": "form-group",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "icon": "",
            "iconAlign": "left",
            "id": "contactNameTextLabel",
            "label": "",
            "labelColon": false,
            "name": "",
            "onChange": [Function],
            "placeholder": "Your Name",
            "size": "",
            "type": "text",
            "value": "name",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Input
            className=""
            icon=""
            iconAlign="left"
            id="contactEmailTextLabel"
            label=""
            labelColon={false}
            name=""
            onChange={[Function]}
            placeholder="Your Email"
            size=""
            type="email"
            value="mail@gmail.com"
          />,
          "className": "form-group",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "icon": "",
            "iconAlign": "left",
            "id": "contactEmailTextLabel",
            "label": "",
            "labelColon": false,
            "name": "",
            "onChange": [Function],
            "placeholder": "Your Email",
            "size": "",
            "type": "email",
            "value": "mail@gmail.com",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Input
            className=""
            icon=""
            iconAlign="left"
            id="contactPhoneTextLabel"
            label=""
            labelColon={false}
            name=""
            onChange={[Function]}
            placeholder="Your Phone"
            size=""
            type="text"
            value="5476"
          />,
          "className": "form-group",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "icon": "",
            "iconAlign": "left",
            "id": "contactPhoneTextLabel",
            "label": "",
            "labelColon": false,
            "name": "",
            "onChange": [Function],
            "placeholder": "Your Phone",
            "size": "",
            "type": "text",
            "value": "5476",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Input
            className=""
            icon=""
            iconAlign="left"
            id="contactMessageTextLabel"
            label=""
            labelColon={false}
            name=""
            onChange={[Function]}
            placeholder="Your Message"
            size=""
            type="textarea"
            value="This is a new message"
          />,
          "className": "form-group",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "icon": "",
            "iconAlign": "left",
            "id": "contactMessageTextLabel",
            "label": "",
            "labelColon": false,
            "name": "",
            "onChange": [Function],
            "placeholder": "Your Message",
            "size": "",
            "type": "textarea",
            "value": "This is a new message",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            active={false}
            ariaLabel=""
            circle={false}
            className=""
            color="default"
            disabled={false}
            loading={false}
            loadingClass="fa fa-circle-o-notch fa-spin"
            onClick={[Function]}
            outline={false}
            rounded={true}
            size=""
            style={Object {}}
            tooltip=""
            type="button"
          >
            Submit
          </Button>,
          "className": "text-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "ariaLabel": "",
            "children": "Submit",
            "circle": false,
            "className": "",
            "color": "default",
            "disabled": false,
            "loading": false,
            "loadingClass": "fa fa-circle-o-notch fa-spin",
            "onClick": [Function],
            "outline": false,
            "rounded": true,
            "size": "",
            "style": Object {},
            "tooltip": "",
            "type": "button",
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormGroup
            className="form-group"
          >
            <Input
              className=""
              icon=""
              iconAlign="left"
              id="contactNameTextLabel"
              label=""
              labelColon={false}
              name=""
              onChange={[Function]}
              placeholder="Your Name"
              size=""
              type="text"
              value="name"
            />
          </FormGroup>,
          <FormGroup
            className="form-group"
          >
            <Input
              className=""
              icon=""
              iconAlign="left"
              id="contactEmailTextLabel"
              label=""
              labelColon={false}
              name=""
              onChange={[Function]}
              placeholder="Your Email"
              size=""
              type="email"
              value="mail@gmail.com"
            />
          </FormGroup>,
          <FormGroup
            className="form-group"
          >
            <Input
              className=""
              icon=""
              iconAlign="left"
              id="contactPhoneTextLabel"
              label=""
              labelColon={false}
              name=""
              onChange={[Function]}
              placeholder="Your Phone"
              size=""
              type="text"
              value="5476"
            />
          </FormGroup>,
          <FormGroup
            className="form-group"
          >
            <Input
              className=""
              icon=""
              iconAlign="left"
              id="contactMessageTextLabel"
              label=""
              labelColon={false}
              name=""
              onChange={[Function]}
              placeholder="Your Message"
              size=""
              type="textarea"
              value="This is a new message"
            />
          </FormGroup>,
          <div
            className="text-center"
          >
            <Button
              active={false}
              ariaLabel=""
              circle={false}
              className=""
              color="default"
              disabled={false}
              loading={false}
              loadingClass="fa fa-circle-o-notch fa-spin"
              onClick={[Function]}
              outline={false}
              rounded={true}
              size=""
              style={Object {}}
              tooltip=""
              type="button"
            >
              Submit
            </Button>
          </div>,
        ],
        "className": "ContactUs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Input
              className=""
              icon=""
              iconAlign="left"
              id="contactNameTextLabel"
              label=""
              labelColon={false}
              name=""
              onChange={[Function]}
              placeholder="Your Name"
              size=""
              type="text"
              value="name"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "",
              "iconAlign": "left",
              "id": "contactNameTextLabel",
              "label": "",
              "labelColon": false,
              "name": "",
              "onChange": [Function],
              "placeholder": "Your Name",
              "size": "",
              "type": "text",
              "value": "name",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Input
              className=""
              icon=""
              iconAlign="left"
              id="contactEmailTextLabel"
              label=""
              labelColon={false}
              name=""
              onChange={[Function]}
              placeholder="Your Email"
              size=""
              type="email"
              value="mail@gmail.com"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "",
              "iconAlign": "left",
              "id": "contactEmailTextLabel",
              "label": "",
              "labelColon": false,
              "name": "",
              "onChange": [Function],
              "placeholder": "Your Email",
              "size": "",
              "type": "email",
              "value": "mail@gmail.com",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Input
              className=""
              icon=""
              iconAlign="left"
              id="contactPhoneTextLabel"
              label=""
              labelColon={false}
              name=""
              onChange={[Function]}
              placeholder="Your Phone"
              size=""
              type="text"
              value="5476"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "",
              "iconAlign": "left",
              "id": "contactPhoneTextLabel",
              "label": "",
              "labelColon": false,
              "name": "",
              "onChange": [Function],
              "placeholder": "Your Phone",
              "size": "",
              "type": "text",
              "value": "5476",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Input
              className=""
              icon=""
              iconAlign="left"
              id="contactMessageTextLabel"
              label=""
              labelColon={false}
              name=""
              onChange={[Function]}
              placeholder="Your Message"
              size=""
              type="textarea"
              value="This is a new message"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "",
              "iconAlign": "left",
              "id": "contactMessageTextLabel",
              "label": "",
              "labelColon": false,
              "name": "",
              "onChange": [Function],
              "placeholder": "Your Message",
              "size": "",
              "type": "textarea",
              "value": "This is a new message",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              active={false}
              ariaLabel=""
              circle={false}
              className=""
              color="default"
              disabled={false}
              loading={false}
              loadingClass="fa fa-circle-o-notch fa-spin"
              onClick={[Function]}
              outline={false}
              rounded={true}
              size=""
              style={Object {}}
              tooltip=""
              type="button"
            >
              Submit
            </Button>,
            "className": "text-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "ariaLabel": "",
              "children": "Submit",
              "circle": false,
              "className": "",
              "color": "default",
              "disabled": false,
              "loading": false,
              "loadingClass": "fa fa-circle-o-notch fa-spin",
              "onClick": [Function],
              "outline": false,
              "rounded": true,
              "size": "",
              "style": Object {},
              "tooltip": "",
              "type": "button",
            },
            "ref": null,
            "rendered": "Submit",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
